digital_asset_types/src/dao/mod.rs

```diff
@@ -245,6 +249,26 @@ impl SearchAssetsQuery {
             joins.push(rel);
         }
 
+        if let Some(n) = self.name.to_owned() {
+            let name_as_str = std::str::from_utf8(&n).map_err(|_| {
+                DbErr::Custom(
+                    "Could not convert raw name bytes into string for comparison".to_owned(),
+                )
+            })?;
+
+            let cond = Condition::all().add(asset_data::Column::RawName.like(name_as_str));
```

Comment:
You might be better off filtering on the name field within the "Metadata" object. RawName doesn't exist on the metaplex version. 

Long story, but the indexed name is stored as a string and removes illegal characters for postgres. The rawname stores the exact name as bytes. We added this since some people need the raw name if they want to reconstruct the metadata args.

Reply:
so there's an `asset_data::Column::Metadata` variant, but how would i get the name and do the comparison from there?

Reply:
`metadata` is a json structure, you would need to extract the `name` from there. Here's a few examples 

```
metadata
"{""name"":""Google Issues Policy Update"",""image"":""https://arweave.net/CnOa23Qj1fHAanXgxQXMxxl0Efx8VCKeAHlLUq5t2QE?ext=jpg"",""symbol"":"""",""attributes"":[{""value"":""Week of 9.12.2023"",""trait_type"":""Date""},{""value"":""$6.3M"",""trait_type"":""Solana NFT Sales Volume""},{""value"":""168K"",""trait_type"":""Solana NFT # of Sales""},{""value"":""$17.92"",""trait_type"":""Price of SOL""},{""value"":""Google to Allow NFT Games Ads"",""trait_type"":""Headline 1""},{""value"":""Monke DAO Votes to Terminate Gen 3 Team Agreement"",""trait_type"":""Headline 2""},{""value"":""Friend.Tech Surpasses $20M TVL"",""trait_type"":""Headline 3""},{""value"":""The Heist"",""trait_type"":""Highest Volume 1""},{""value"":""VTOPIANS"",""trait_type"":""Highest Volume 2""},{""value"":""Transdimensional Fox Federation"",""trait_type"":""Highest Volume 3""},{""value"":""Common"",""trait_type"":""Rarity""}],""properties"":{""files"":[{""uri"":""https://arweave.net/CnOa23Qj1fHAanXgxQXMxxl0Efx8VCKeAHlLUq5t2QE?ext=jpg"",""type"":""image/jpeg""}],""category"":""image""},""description"":""Google to Allow NFT Games Ads • Monke DAO Votes to Terminate Gen 3 Team Agreement • Friend.Tech Surpasses $20M TVL"",""external_url"":"""",""animation_url"":""""}"
"{""name"":""Dream Walker"",""image"":""https://arweave.net/2M-ZU2R9m88XU1ZqZSNfB6S8S1LhWkwCjtMtItPAwHs?ext=jpg"",""symbol"":"""",""attributes"":[{""value"":""2"",""trait_type"":""Season""},{""value"":""@M0TZS"",""trait_type"":""Artist""},{""value"":""17"",""trait_type"":""Drop""},{""value"":""1"",""trait_type"":""Variation""},{""value"":""Common"",""trait_type"":""Rarity""}],""properties"":{""files"":[{""uri"":""https://arweave.net/2M-ZU2R9m88XU1ZqZSNfB6S8S1LhWkwCjtMtItPAwHs?ext=jpg"",""type"":""image/jpeg""}],""category"":""image""},""description"":""The Cosmic Occult of Motz."",""external_url"":"""",""animation_url"":""""}"
"{""name"":""Eyes Wide Shit"",""image"":""https://arweave.net/FOUgn4wzzbuTdMlL5WTA1iodM2Y1ZbWhKRZOTMEIgKs?ext=jpeg"",""symbol"":"""",""attributes"":[{""value"":""1"",""trait_type"":""Season""},{""value"":""8"",""trait_type"":""Drop""},{""value"":""Common"",""trait_type"":""Rarity""}],""properties"":{""files"":[{""uri"":""https://arweave.net/FOUgn4wzzbuTdMlL5WTA1iodM2Y1ZbWhKRZOTMEIgKs?ext=jpeg"",""type"":""image/jpeg""}],""category"":""image""},""description"":""The Dirty Dozen? A cult? Of course not..."",""external_url"":"""",""animation_url"":""""}"
"{""name"":""Breaking: SEC Sues Coinbase"",""image"":""https://arweave.net/Th5WutzqZLzzEW46eazvsOaMf7ElDgGKO9KCgmS6y50?ext=png"",""symbol"":"""",""attributes"":[{""value"":""Week of 6.13.2023"",""trait_type"":""Date""},{""value"":""$8.3M"",""trait_type"":""Solana NFT Sales Volume""},{""value"":""118k"",""trait_type"":""Solana NFT # of Sales""},{""value"":""14.6k"",""trait_type"":""Solana NFT # of Buyers""},{""value"":""$14.94"",""trait_type"":""Price of SOL""},{""value"":""SEC Sues Coinbase and Binance"",""trait_type"":""Headline 1""},{""value"":""Apple Announces VR Headset Vision Pro"",""trait_type"":""Headline 2""},{""value"":""Robinhood Delists $SOL"",""trait_type"":""Headline 3""},{""value"":""Mad Lads"",""trait_type"":""Highest Volume 1""},{""value"":""Zero Monke Biz"",""trait_type"":""Highest Volume 2""},{""value"":""ABC"",""trait_type"":""Highest Volume 3""},{""value"":""Common"",""trait_type"":""Rarity""}],""properties"":{""files"":[{""uri"":""https://arweave.net/Th5WutzqZLzzEW46eazvsOaMf7ElDgGKO9KCgmS6y50?ext=png"",""type"":""image/png""}],""category"":""image""},""description"":""SEC Sues Coinbase and Binance • Apple Announces VR Headset Vision Pro • Robinhood Delists $SOL"",""external_url"":"""",""animation_url"":""""}"
"{""name"":""Fireworks"",""image"":""https://arweave.net/SS80YVoWJpR_Jl62lSzuTJjHan-vBn2SpPbLS9gZnh4?ext=jpg"",""symbol"":"""",""attributes"":[{""value"":""13"",""trait_type"":""drop""},{""value"":""2023-07-10"",""trait_type"":""created""},{""value"":""mp4"",""trait_type"":""filetype""},{""value"":""1500 x 1500"",""trait_type"":""dimensions""},{""value"":""square"",""trait_type"":""orientation""},{""value"":""common"",""trait_type"":""rarity""}],""properties"":{""files"":[{""uri"":""https://arweave.net/SS80YVoWJpR_Jl62lSzuTJjHan-vBn2SpPbLS9gZnh4?ext=jpg"",""type"":""image/jpeg""},{""uri"":""https://arweave.net/vBYeFx28WgVIwP2lMadoFgxW5rUYbg-hNOFDKQNHl-A?ext=mp4"",""type"":""video/mp4""}],""category"":""video""},""description"":""A hand drawn animation set to the sounds of fireworks recorded near Degen Poet’s home."",""external_url"":"""",""animation_url"":""https://arweave.net/vBYeFx28WgVIwP2lMadoFgxW5rUYbg-hNOFDKQNHl-A?ext=mp4""}"
"{""name"":""Dream Walker"",""image"":""https://arweave.net/2M-ZU2R9m88XU1ZqZSNfB6S8S1LhWkwCjtMtItPAwHs?ext=jpg"",""symbol"":"""",""attributes"":[{""value"":""2"",""trait_type"":""Season""},{""value"":""@M0TZS"",""trait_type"":""Artist""},{""value"":""17"",""trait_type"":""Drop""},{""value"":""1"",""trait_type"":""Variation""},{""value"":""Common"",""trait_type"":""Rarity""}],""properties"":{""files"":[{""uri"":""https://arweave.net/2M-ZU2R9m88XU1ZqZSNfB6S8S1LhWkwCjtMtItPAwHs?ext=jpg"",""type"":""image/jpeg""}],""category"":""image""},""description"":""The Cosmic Occult of Motz."",""external_url"":"""",""animation_url"":""""}"
"{""name"":""Chapter Five"",""image"":""https://arweave.net/EqpJI2JNvNminSVzEtg8e-vep_4HWmQaGAnafhFAcq4?ext=jpg"",""symbol"":"""",""attributes"":[{""value"":""Five"",""trait_type"":""Chapter""},{""value"":""July 18, 2023"",""trait_type"":""Date""},{""value"":""Foursixsix"",""trait_type"":""Written By""},{""value"":""Foursixsix"",""trait_type"":""Cover By""},{""value"":""Foursixsix"",""trait_type"":""Interiors By""},{""value"":""4"",""trait_type"":""Story Pages""},{""value"":""https://drip.haus/bork, https://drip-tracker.vercel.app"",""trait_type"":""Readable On""},{""value"":""Common"",""trait_type"":""Rarity""}],""properties"":{""files"":[{""uri"":""https://arweave.net/EqpJI2JNvNminSVzEtg8e-vep_4HWmQaGAnafhFAcq4?ext=jpg"",""type"":""image/jpeg""},{""uri"":""https://arweave.net/eRv6VqMZQqUUlS-iNi2RQtn64XYWWG6kJFyDvgpdHNk?ext=jpg#slide_1"",""type"":""image/jpeg""},{""uri"":""https://arweave.net/aFMCMkyAH-HPnkNLO8EPDZxjRG8M0LFN8W7urztwbkk?ext=jpg#slide_2"",""type"":""image/jpeg""},{""uri"":""https://arweave.net/HddyIhJd5SW8BbOBhhIzlBZ2O0HmdolXBQQ4cbbtXK4?ext=jpg#slide_3"",""type"":""image/jpeg""},{""uri"":""https://arweave.net/2V7tspiD0m8SBKbuntbR3YBM-42vGVWMPe7OVMlH-nA?ext=jpg#slide_4"",""type"":""image/jpeg""},{""uri"":""https://arweave.net/DLkxKfgdqSSKYP_DdB9BgCt9h773xnCu6qc8gArzJPI?ext=jpg#slide_5"",""type"":""image/jpeg""}],""category"":""image""},""description"":""Bork’s dreams turn into nightmares."",""external_url"":"""",""animation_url"":""""}"
"{""name"":""TipLink Compressed #836655"",""image"":""https://shdw-drive.genesysgo.net/4mWjD7horGWeG5EvaV9fYWmdjQt2av9984z7JVFteaQ6/30-106.png"",""symbol"":""cTIPLINK"",""attributes"":[{""value"":""LowPoly"",""trait_type"":""Style""},{""value"":""3D Damask Pattern"",""trait_type"":""Subject""},{""value"":""Stable Diffusion 2.1"",""trait_type"":""Model""},{""value"":""TipLink"",""trait_type"":""Created with 💙 by:""}],""properties"":{""files"":[{""uri"":""https://shdw-drive.genesysgo.net/4mWjD7horGWeG5EvaV9fYWmdjQt2av9984z7JVFteaQ6/30-106.png"",""type"":""image/png""}],""dreamStudio"":{""seed"":106,""steps"":50,""style"":""lowpoly"",""prompt"":""2840cb1808022febb7aaf6436bff90a47755adeda668bc5f952c09b3434317ae"",""strength"":7,""negativePrompt"":""Staged apartments""}},""description"":""NFTs created on Solana using account compression. There are limitless NFTs in this collection, so don't speculate!"",""seller_fee_basis_points"":0}"
"{""name"":""Fly"",""image"":""https://arweave.net/76Iv55pUrj7CNdTL8oa-HorX6fQw-i4dnVdZMbZr5g0?ext=gif"",""symbol"":"""",""attributes"":[{""value"":""6"",""trait_type"":""drop""},{""value"":""2023-05-22"",""trait_type"":""created""},{""value"":""gif"",""trait_type"":""filetype""},{""value"":""1200 x 675"",""trait_type"":""dimensions""},{""value"":""landscape"",""trait_type"":""orientation""},{""value"":""common"",""trait_type"":""rarity""}],""properties"":{""files"":[{""uri"":""https://arweave.net/76Iv55pUrj7CNdTL8oa-HorX6fQw-i4dnVdZMbZr5g0?ext=gif"",""type"":""image/gif""}],""category"":""image""},""description"":""Animated GIF of Link jumping off a cliff. Created by Degen Poet using watercolor pencils + an Underwood Quiet Tab typewriter. Based on a scene from Zelda Tears of the Kingdom."",""external_url"":"""",""animation_url"":""""}"
````

Reply:
Hi, this is patched internally and deployed to production. Can you test if it is working as expected?